src/emotes.ts

```diff
@@ -0,0 +1,107 @@
+import { emotes } from './emote_list';
+
+/* -------------------- HELPER FUNCTIONS -------------------- */
+
+const parseMessage = (text: string, HTML: string) => {
+  const emotesInMessage = [];
+  const messageString = text.split(' ');
+
+  // determine which emotes are within the message
+  emotes.forEach(({ name, id }) => {
+    if (text.includes(name)) {
+      emotesInMessage.push({ name, id });
+    }
+  });
+
+  // compare each word in the message with the emotes, if there is an exact match, replace them
+  messageString.forEach((word: string) => {
```

Comment:
small nit: if `messageString` is defined as an array of strings, we won't have to define `word`'s type as `string` snce it will be implied